Issues with Keycloak and Web Integration

I'm using an existing keycloak instance and running into the error below after authenticating:

![key](https://github.com/user-attachments/assets/77010413-6b07-47d4-abfa-f35095a7afda)


Keycloak integration code:


settings.yml
```
auth:
  providers:
    - keycloak
```

.env
```
KEYCLOAK_PUBLIC_URL="https://keycloak.mydomain.domain"
KEYCLOAK_REALM="xxxx"
KEYCLOAK_SECRET="xxxxxxxx"
KEYCLOAK_CLIENT_ID="xxxx"
```


Web integration code:

```
web:
  engines:
    - provider: "searxng"
    - engine: "pubmed"
    - use_pdf: true  # Some SearXNG engines give PDF URLs - this tells Abbey to go to the PDF rather than the regular result
searxng:
  url: "http://searxng:8080"  # Replace with your URL
```

When trying to add the web integration, the backend and celery fail with the error logs below:

backend-1    | [2024-12-19 04:18:15 +0000] [8] [ERROR] Exception in worker process
backend-1    | Traceback (most recent call last):
backend-1    |   File "/usr/local/lib/python3.8/dist-packages/gunicorn/arbiter.py", line 608, in spawn_worker
backend-1    |     worker.init_process()
backend-1    |   File "/usr/local/lib/python3.8/dist-packages/gunicorn/workers/ggevent.py", line 146, in init_process
backend-1    |     super().init_process()
backend-1    |   File "/usr/local/lib/python3.8/dist-packages/gunicorn/workers/base.py", line 135, in init_process
backend-1    |     self.load_wsgi()
backend-1    |   File "/usr/local/lib/python3.8/dist-packages/gunicorn/workers/base.py", line 147, in load_wsgi
backend-1    |     self.wsgi = self.app.wsgi()
backend-1    |   File "/usr/local/lib/python3.8/dist-packages/gunicorn/app/base.py", line 66, in wsgi
backend-1    |     self.callable = self.load()
backend-1    |   File "/usr/local/lib/python3.8/dist-packages/gunicorn/app/wsgiapp.py", line 57, in load
backend-1    |     return self.load_wsgiapp()
backend-1    |   File "/usr/local/lib/python3.8/dist-packages/gunicorn/app/wsgiapp.py", line 47, in load_wsgiapp
backend-1    |     return util.import_app(self.app_uri)
backend-1    |   File "/usr/local/lib/python3.8/dist-packages/gunicorn/util.py", line 423, in import_app
backend-1    |     app = app(*args, **kwargs)
backend-1    |   File "/app/app/__init__.py", line 13, in create_app
backend-1    |     from .configs.user_config import BACKEND_VERSION
backend-1    |   File "/app/app/configs/user_config.py", line 5, in <module>
backend-1    |     from ..integrations.web import SEARCH_PROVIDERS
backend-1    |   File "/app/app/integrations/web.py", line 149, in <module>
backend-1    |     SEARCH_PROVIDERS = generate_engines()
backend-1    |   File "/app/app/integrations/web.py", line 129, in generate_engines
backend-1    |     provider = option['provider']
backend-1    | KeyError: 'provider'
backend-1    | [2024-12-19 04:18:15 +0000] [8] [INFO] Worker exiting (pid: 8)
celery-1     | Usage: celery [OPTIONS] COMMAND [ARGS]...
celery-1     | Try 'celery --help' for help.
celery-1     | 
celery-1     | Error: Invalid value for '-A' / '--app': 
celery-1     | Unable to load celery application.
celery-1     | While trying to load the module app.worker.celery the following error occurred:
celery-1     | Traceback (most recent call last):
celery-1     |   File "/usr/local/lib/python3.8/dist-packages/celery/bin/celery.py", line 58, in convert
celery-1     |     return find_app(value)
celery-1     |   File "/usr/local/lib/python3.8/dist-packages/celery/app/utils.py", line 383, in find_app
celery-1     |     sym = symbol_by_name(app, imp=imp)
celery-1     |   File "/usr/local/lib/python3.8/dist-packages/kombu/utils/imports.py", line 59, in symbol_by_name
celery-1     |     module = imp(module_name, package=package, **kwargs)
celery-1     |   File "/usr/local/lib/python3.8/dist-packages/celery/utils/imports.py", line 109, in import_from_cwd
celery-1     |     return imp(module, package=package)
celery-1     |   File "/usr/lib/python3.8/importlib/__init__.py", line 127, in import_module
celery-1     |     return _bootstrap._gcd_import(name[level:], package, level)
celery-1     |   File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
celery-1     |   File "<frozen importlib._bootstrap>", line 991, in _find_and_load
celery-1     |   File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
celery-1     |   File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
celery-1     |   File "<frozen importlib._bootstrap_external>", line 848, in exec_module
celery-1     |   File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
celery-1     |   File "/app/app/worker.py", line 4, in <module>
celery-1     |     from .retriever import Retriever, ResourceRetriever
celery-1     |   File "/app/app/retriever.py", line 3, in <module>
celery-1     |     from .auth import User
celery-1     |   File "/app/app/auth.py", line 4, in <module>
celery-1     |     from .configs.user_config import PERMISSIONING
celery-1     |   File "/app/app/configs/user_config.py", line 5, in <module>
celery-1     |     from ..integrations.web import SEARCH_PROVIDERS
celery-1     |   File "/app/app/integrations/web.py", line 149, in <module>
celery-1     |     SEARCH_PROVIDERS = generate_engines()
celery-1     |   File "/app/app/integrations/web.py", line 129, in generate_engines
celery-1     |     provider = option['provider']
celery-1     | KeyError: 'provider'
celery-1     | 
backend-1    | [2024-12-19 04:18:16 +0000] [1] [ERROR] Worker (pid:8) exited with code 3
backend-1    | [2024-12-19 04:18:16 +0000] [1] [ERROR] Shutting down: Master
backend-1    | [2024-12-19 04:18:16 +0000] [1] [ERROR] Reason: Worker failed to boot.
backend-1 exited with code 3
celery-1 exited with code 2

--- a/README.md
+++ b/README.md
@@ -284,8 +284,8 @@ web:
     - engine: "pubmed"  # Only used for SearXNG - leave blank to search over all engines you've enabled
     
     - provider: "searxng"
-    - engine: "arxiv"
-    - use_pdf: true  # Some SearXNG engines give PDF URLs - this tells Abbey to go to the PDF rather than the regular result
+      engine: "arxiv"
+      use_pdf: true  # Some SearXNG engines give PDF URLs - this tells Abbey to go to the PDF rather than the regular result
 ```
 
 | Provider   | Provider Code | API Key Name           | Needs Provider Setting |

--- a/frontend/src/config/generateEnv.js
+++ b/frontend/src/config/generateEnv.js
@@ -66,7 +66,9 @@ try {
                 else if (provider == 'keycloak'){
                     addEnv('NEXT_PUBLIC_ENABLE_KEYCLOAK_AUTH', 1)
                     addEnv('KEYCLOAK_PUBLIC_URL', env['KEYCLOAK_PUBLIC_URL'])
-                    addEnv('KEYCLOAK_INTERNAL_URL', env['KEYCLOAK_INTERNAL_URL'])
+                    if (env['KEYCLOAK_INTERNAL_URL']){
+                        addEnv('KEYCLOAK_INTERNAL_URL', env['KEYCLOAK_INTERNAL_URL'])
+                    }
                     addEnv('KEYCLOAK_REALM', env['KEYCLOAK_REALM'])
                     addEnv('KEYCLOAK_SECRET', env['KEYCLOAK_SECRET'])
                     addEnv('KEYCLOAK_CLIENT_ID', env['KEYCLOAK_CLIENT_ID'])

